### PR TITLE
Replace LCD with REST

### PR DESCRIPTION
**Pull request by joon9823**
- replace LCD with REST
- but still leave LCDClient for backward compatibility

## Summary by CodeRabbit

- **New Features**
	- Transitioned from using an LCD client to a REST client for improved API interactions.
	- Updated usage examples and documentation to reflect the new REST client terminology.

- **Bug Fixes**
	- Adjusted API request URLs in various test files to point to the new REST endpoint.

- **Documentation**
	- Renamed references in the README and updated descriptions to align with the REST client changes.

- **Refactor**
	- Updated class structures and method implementations across multiple files to utilize the REST client.

- **Tests**
	- Modified test cases to ensure compatibility with the new REST client and updated base URLs accordingly.

### CODE REVIEW

**Review by Vritra4 — APPROVED** [review]
LGTM

**Comment by Vritra4** [comment]
please mention about it on releasenote

File: src/client/rest/api/BankAPI.spec.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { APIRequester } from '../APIRequester'
 import { BankAPI } from './BankAPI'
 
-const c = new APIRequester('https://lcd.devnet.initia.xyz/')
+const c = new APIRequester('https://rest.devnet.initia.xyz/')
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/client/rest/api/BankAPI.spec.ts:4]: this is for backward compatibility

File: src/client/rest/api/IbcAPI.spec.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { APIRequester } from '../APIRequester'
 import { IbcAPI } from './IbcAPI'
 
-const c = new APIRequester('https://lcd.devnet.initia.xyz/')
+const c = new APIRequester('https://rest.devnet.initia.xyz/')
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/client/rest/api/IbcAPI.spec.ts:4]: not an issue

File: src/client/rest/RESTClient.ts [hunk 3]
```diff
@@ -172,3 +172,6 @@ export class LCDClient {
     return new Wallet(this, key)
   }
 }
+
+// For backward compatibility
+export class LCDClient extends RESTClient {}
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/client/rest/RESTClient.ts]: fixed

File: src/client/rest/RESTClient.ts [hunk 4]
```diff
@@ -63,23 +63,23 @@ const DEFAULT_GAS_PRICES_BY_CHAIN_ID: Record<string, Coins.Input> = {
 }
 
 /**
- * An object repesenting a connection to a initiad node running the Lite Client Daemon (LCD)
+ * An object repesenting a connection to a initiad node running the REST Client
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/client/rest/RESTClient.ts]: fixed

File: src/client/rest/RESTClient.ts [hunk 5]
```diff
@@ -63,23 +63,23 @@ const DEFAULT_GAS_PRICES_BY_CHAIN_ID: Record<string, Coins.Input> = {
...
  * ```ts
- * import { LCDClient } from 'initia.js';
+ * import { RESTClient } from 'initia.js';
  *
- * const initia = new LCDClient("https://lcd.devnet.initia.xyz", {
+ * const initia = new RESTClient("https://rest.devnet.initia.xyz", {
  *    chainId: "testnet"
  * });
  * ```
  */
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/client/rest/RESTClient.ts:78]: check yourself again, looks the same to me

**Pull request merged**